codex/raw/marketplace.md

```diff
@@ -0,0 +1,275 @@
+| slots | Created by client nodes when a new dataset is requested to be stored. Discussed further in the [slots section](#slots).  |
+
+### Storage Request
+
+Client nodes can create storage requests on the Codex network via the Codex marketplace.
+The marketplace is a smart contract deployed on a EVM-compatiable blockchain.
+The smart contract MUST handle storage request, storage slot state, 
+storage provider rewards, storage provider collateral, and storage proof state. 
+
+To create a request to store a dataset on the Codex network,
+client nodes MUST split the dataset into data chunks, $(c_1, c_2, c_3, \ldots, c_{n})$.
+Using an erasure coding technique, 
+the data chunks are encoded and placed into separate slots.
+The erasure coding technique SHOULD be the [Reed-Soloman algorithm](https://hackmd.io/FB58eZQoTNm-dnhu0Y1XnA).
+
+The requester, client node, SHOULD submit a transaction with the desired request parameters.
+The requester MUST provide a `duration` value for the storage request along with the appropriate `reward`,
+payment for request. 
+Once a request is created via the transaction, 
+all slots MUST be filled by storage providers before the request is officially started.
+If the request does not attract enough storage providers after a time set by the requester,
+which is defined by as `expiry`,
+the request MUST be `canceled`.
+If canceled, 
+`collateral` SHOULD be returned to all storage providers that participated in the initiation of the request, 
+including a portion of the `reward`. 
+The remaining `reward` is returned to the requester.
+The requester MAY create a new request with different values to restart the process.
+
+The storage request SHOULD initiate a new storage contract with the following values:
+
+```solidity
+
+  // the Codex node requesting storage
+  address client;
+
+  // content identifier
+  string cid;
+
+  // merkle root of the dataset, used to verify storage proofs
+  byte32 merkleRoot;
+
+  // amount of token from the requester to reward storage providers
+  uint256 reward;
+
+  // amount of tokens required for collateral by storage providers
+  uint256 collateral;
+
+  // frequency that proofs are checked by validator nodes
+  uint256 proofProbability;
+
+  // amount of desired time for stoage request
+  uint256 duration;
+
+  // the number of requested slots
+  uint64 slots;
+
+  // amount of storage per slot
+  uint256 slotSize;
+
+  // Amount of time before request expires
+  uint256 expiry;
+
+  // random value to differentiate from other requests
+  byte32 nonce;
+
+```
+
+`cid` 
+
+An identifier used to locate the dataset
+- MUST be a sha-256 hash, [CIDv1](https://github.com/multiformats/cid#cidv1)
+- MUST be generated by the client node
+
+`reward`
+
+- it is an REQUIRED amount to be included in the transaction for a storage request.
+- it SHOULD be amount of tokens offered per byte per second
```

Comment:
I am a bit confused by the "SHOULD" here. We do not offer any other pricing mechanism, like "reward for the whole storage request" or something like that. Should it be therefore really "SHOULD"?
```suggestion
- it MUST be amount of tokens offered per byte per second
```

Reply:
I suggested SHOULD as the user is free to enter any amount with the request transaction. If Codex marketplace rejects a storage request transaction because the reward was not sufficient then it can be a MUST.

Reply:
I see, okey, agree 👍